sds_data_manager/stacks/sqs_stack.py

```diff
@@ -0,0 +1,75 @@
+"""Stack for creating instrument SQS queues and attaching batch_starter as a target."""
+
+from aws_cdk import Duration, Stack, aws_sqs
+from aws_cdk import aws_events as events
+from aws_cdk import aws_events_targets as targets
+from constructs import Construct
+
+
+class SqsStack(Stack):
+    """Stack to create instrument/level SQS queues and attach them to EventBridge."""
+
+    def __init__(
+        self,
+        scope: Construct,
+        construct_id: str,
+        instrument_names: list[str],
+        **kwargs,
+    ):
+        """Create a stack to create a SQS queue and Eventbridge rule for an instrument.
+
+        Parameters
+        ----------
+        scope : Construct
+            Parent construct.
+        construct_id : str
+            A unique string identifier for this construct.
+        instrument_names : list[str]
+            A list of all instrument names
+        kwargs : dict
+            Keyword arguments
+        """
+        super().__init__(scope, construct_id, **kwargs)
+        # This needs to be a FIFO queue to enforce ordering
+        self.instrument_queue = aws_sqs.Queue(
+            self,
+            "FileArrivalQueue",
+            queue_name="file_arrival_queue.fifo",
+            fifo=True,
+            encryption=aws_sqs.QueueEncryption.UNENCRYPTED,
+            # This timeout determines how long the queue waits for processing. It must
+            # be longer than the timeout of the lambda.
+            visibility_timeout=Duration.seconds(300),
```

Comment:
No, we aren't waiting 5 minutes for a file arrives. It's a timeout for how long the queue will wait before moving the event to the dead letter queue, I believe. I made it long just to accommodate any changes in the future to that lambda - it only comes into play if processing is stalled for some reason, so I made it longer just in case. In most cases, processing takes a second or two and the events are removed from the queue that quickly.